citesphere-model/src/main/java/edu/asu/diging/citesphere/data/bib/impl/CitationMongoDao.java

```diff
@@ -71,5 +83,22 @@ public List<? extends ICitation> findCitationsInCollection(String groupId, Strin
         query.limit(pageSize);
         return mongoTemplate.find(query, Citation.class);
     }
-    
-}
+    @Override
+    public Citations findCitatationByName(String name){
+    	name=" "+name;
+        ProjectionOperation project = Aggregation.project().and("authors")
+                .unionArrays("editors", "otherCreators.person").as("persons").and(Aggregation.ROOT).as("citation");
+        
+        UnwindOperation unwind = Aggregation.unwind("persons");
+
+        MatchOperation match = Aggregation.match(Criteria.where("persons.name").is(name));
+
+        GroupOperation group = Aggregation.group("persons.name").push("citation").as("citations");
+
+        
+        AggregationResults<Citations> result = mongoTemplate.aggregate(
+                Aggregation.newAggregation(project, unwind, match, group),
+                Citation.class, Citations.class);
+
+        return result.getUniqueMappedResult();
```

Comment:
citations should use the URI of an author (the name can be the same for two different people, uris are unique).

Reply:
Please check with Pratik since he needs the same thing.